code/modules/vehicles/armored/armored_weapons.dm

```diff
@@ -0,0 +1,314 @@
+
+/// do after checks for the mecha equipment do afters
+/obj/item/armored_weapon/proc/do_after_checks(atom/target)
+	if(!chassis)
+		return FALSE
+	if(QDELETED(current_target))
+		return FALSE
+	if(chassis.primary_weapon == src)
+		var/dir_target_diff = get_between_angles(Get_Angle(chassis, current_target), dir2angle(chassis.turret_overlay.dir))
+		if(dir_target_diff > (ARMORED_FIRE_CONE_ALLOWED / 2))
+			if(!chassis.swivel_turret(current_target))
+				return FALSE
+		dir_target_diff = get_between_angles(Get_Angle(chassis, current_target), dir2angle(chassis.turret_overlay.dir))
+		if(dir_target_diff > (ARMORED_FIRE_CONE_ALLOWED / 2))
+			return FALSE
+	return TRUE
+
+///callback wrapper for adding/removing trait
+/obj/item/armored_weapon/proc/set_bursting(bursting)
+	if(bursting)
+		ADD_TRAIT(src, TRAIT_GUN_BURST_FIRING, VEHICLE_TRAIT)
+		return
+	REMOVE_TRAIT(src, TRAIT_GUN_BURST_FIRING, VEHICLE_TRAIT)
+
+///Changes the current target.
+/obj/item/armored_weapon/proc/change_target(datum/source, atom/src_object, atom/over_object, turf/src_location, turf/over_location, src_control, over_control, params)
+	SIGNAL_HANDLER
+	set_target(get_turf_on_clickcatcher(over_object, source, params))
+
+///Sets the current target and registers for qdel to prevent hardels
+/obj/item/armored_weapon/proc/set_target(atom/object)
+	if(object == current_target || object == chassis)
+		return
+	if(current_target)
+		UnregisterSignal(current_target, COMSIG_QDELETING)
+	current_target = object
+	if(current_target)
+		RegisterSignal(current_target, COMSIG_QDELETING, PROC_REF(clean_target))
+
+///Stops the Autofire component and resets the current cursor.
+/obj/item/armored_weapon/proc/stop_fire(mob/living/source, atom/object, location, control, params)
+	SIGNAL_HANDLER
+	var/list/modifiers = params2list(params)
+	if(!((modifiers[BUTTON] == RIGHT_CLICK) && chassis.secondary_weapon == src) && !((modifiers[BUTTON] == LEFT_CLICK) && chassis.primary_weapon == src))
+		return
+	SEND_SIGNAL(src, COMSIG_ARMORED_STOP_FIRE)
+	if(!HAS_TRAIT(src, TRAIT_GUN_BURST_FIRING))
+		reset_fire()
+	UnregisterSignal(source, list(COMSIG_MOB_MOUSEDRAG, COMSIG_MOB_MOUSEUP))
+
+///Cleans the current target in case of Hardel
+/obj/item/armored_weapon/proc/clean_target()
+	SIGNAL_HANDLER
+	current_target = get_turf(current_target)
+
+///Resets the autofire component.
+/obj/item/armored_weapon/proc/reset_fire()
+	windup_checked = WEAPON_WINDUP_NOT_CHECKED
+	current_firer?.client?.mouse_pointer_icon = chassis.mouse_pointer
+	set_target(null)
+	current_firer = null
+
+///does any effects and changes to the projectile when it is fired
+/obj/item/armored_weapon/proc/apply_weapon_modifiers(obj/projectile/projectile_to_fire, mob/firer)
+	projectile_to_fire.shot_from = src
+	if(chassis.hitbox?.tank_desants)
+		projectile_to_fire.hit_atoms += chassis.hitbox.tank_desants
+	if(projectile_to_fire.ammo.flags_ammo_behavior & AMMO_IFF)
+		var/iff_signal
```

Comment:
Do you even need a var? if we don't have an iff sig we can pull from the firer, its just going to be null anyway, so we can just set that directly

Reply:
changed